### ISSUE TITLE
Error types

### ISSUE THREAD

**Issue by kgryte** [twada/type-name#4]
Provided an error type other than `Error`, such as `TypeError`, `SyntaxError`, etc,

``` javascript
var err = new TypeError( 'beep' );
typeName( err ) === 'Error';
```

which is confirmed in the module tests: https://github.com/twada/type-name/blob/master/test/test.js#L62. 

Any particular reason not to support returning the actual constructor names: `'TypeError'`, `'SyntaxError'`, etc?

The line in question is https://github.com/twada/type-name/blob/master/test/test.js#L62 which checks for `'Object'`. For an `error` class, this is `'Error'`.

To support returning the `error` classes, this line could be amended to read:

``` javascript
if ((strName === 'Object' || strName === 'Error') && obj.constructor) {
  return funcName(obj.constructor);
}
```

**Comment by twada** [comment]
@kgryte Thank you for your suggestion! It makes sense and makes `type-name` much more useful.
I'll try to apply your suggestion and cut a release as 2.0.0 since it is a breaking change.

**Comment by kgryte** [comment]
@twada Awesome. Thanks for the response!

**Comment by kgryte** [comment]
+1.

**Comment by kgryte** [comment]
Still needs to be published to NPM, correct?

**Comment by twada** [comment]
> Still needs to be published to NPM, correct?

Yes. I'll ship it soon with a few more breaking changes.
So if you are in hurry, please use master branch directly.

**Comment by twada** [comment]
Just released [type-name 2.0.0](https://github.com/twada/type-name/blob/master/CHANGELOG.md#200-2016-01-06) :tada:

**Comment by kgryte** [comment]
Awesome. Thanks!
